torch_geometric/explain/algorithm/base.py

```diff
@@ -37,6 +38,11 @@ def forward(
             target (torch.Tensor): the target of the model.
             target_index (int or torch.Tensor, optional): The target indices to
```

Comment:
Is my understanding correct that we need this only for multi-task learning?

Reply:
yes, I'm just unsure if this is a good way to do it. I'm a bit lost with the multi-task classification setting

Reply:
Ah I see. Lets mention in the doc that this is only needed for multi-task learning. 
Also i don't think we have an actual example of multi-task learning on graphs, so i wouldn't worry about it till someone comes up with an example an then uses `GNNExpaliner` on it.

Reply:
Well now that I think about it, it is also usefull if suppose you have $GNN: (A,X) \mapsto \mathbb{R}^K$ and you are interested in what is important for the $k ^{th}$ coordinate of your output, you'd set _target_index_ to $k$.

Reply:
Ah I see. In that case can me mention this in the doc. Pretty much let the user know that they need to use this if their model outputs a tensor of values, and can ignore it otherwise.